src/AdoNet/Orleans.AdoNet.Core/Orleans.AdoNet.Core.csproj

```diff
@@ -0,0 +1,32 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <PackageId>Microsoft.Orleans.AdoNet.Core</PackageId>
+    <Title>Microsoft Orleans ADO.NET shared components</Title>
+    <Description>Microsoft Orleans shared components for ADO.NET providers</Description>
+    <PackageTags>$(PackageTags) ADO.NET SQL</PackageTags>
+    <TargetFrameworks>$(DefaultTargetFrameworks)</TargetFrameworks>
+    <OrleansBuildTimeCodeGen>true</OrleansBuildTimeCodeGen>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+  </PropertyGroup>
+
+  <PropertyGroup>
+    <AssemblyName>Orleans.AdoNet.Core</AssemblyName>
+    <RootNamespace>Orleans.AdoNet.Core</RootNamespace>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <InternalsVisibleTo Include="Orleans.Clustering.AdoNet"/>
+    <InternalsVisibleTo Include="Orleans.Persistence.AdoNet"/>
+    <InternalsVisibleTo Include="Orleans.Reminders.AdoNet"/>
+    <InternalsVisibleTo Include="Orleans.Streaming.AdoNet"/>
```

Comment:
This requires that the API surface for this project won't change, unless developers are careful to always upgrade these packages in-sync with each other. It doesn't necessarily require changing, but it's worth noting that this could introduce a paper cut for consumers.

Reply:
This stuff is here because the code is still tightly coupled. This will go away on the next clean up, as the services are decoupled into DI. I didn't want to include that effort now or the diff for this PR would become even more unreadable.

Reply:
What's the plan to get rid of these dependencies?